core/src/main/scala/scalafix/rewrite/Xor2Either.scala

```diff
@@ -0,0 +1,55 @@
+package scalafix.rewrite
+
+import scala.collection.immutable.Seq
+import scala.meta.parsers.Parse
+import scala.{meta => m}
+import scalafix.util._
+import scala.meta._
+
+class Desugared[T <: Tree: Parse](implicit semantic: SemanticApi) {
+  def unapply(original: T): Option[T] = semantic.desugared(original)
+}
+
+case object Xor2Either extends Rewrite {
+  override def rewrite(ast: m.Tree, ctx: RewriteCtx): Seq[Patch] = {
+    implicit val semanticApi: SemanticApi = getSemanticApi(ctx)
+    object DType extends Desugared[Type]
```

Comment:
I think we can move these to the `SemanticApi` trait instead of copying them around.